pkg/statsdpublisher/publisher.go

```diff
@@ -0,0 +1,92 @@
+package statsdpublisher
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"os"
+
+	"github.com/canaryio/canary/pkg/sensor"
+)
+
+// Publisher implements canary.Publisher
+type Publisher struct {
+	conn   net.Conn
+	Source string
+	Prefix string
+}
+
+func New(source, host, prefix string) (*Publisher, error) {
+	conn, err := net.Dial("udp", host)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Publisher{
+		conn:   conn,
+		Source: source,
+		Prefix: prefix,
+	}, nil
+}
+
+func NewFromEnv() (*Publisher, error) {
+	var err error
+	source := os.Getenv("SOURCE")
+	if source == "" {
+		source, err = os.Hostname()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	host := os.Getenv("STATSD_HOST")
+	if host == "" {
+		return nil, fmt.Errorf("STATSD_HOST not set in ENV")
+	}
+
+	prefix := os.Getenv("STATSD_PREFIX")
+	if prefix == "" {
+		prefix = "canary"
+	}
+
+	return New(source, host, prefix)
+}
+
+func (p *Publisher) Publish(m sensor.Measurement) error {
+	return Write(p.conn, p.Source, p.Prefix, m)
+}
+
+func Write(w io.Writer, source, prefix string, m sensor.Measurement) error {
```

Comment:
with this, there appears no way to graph failures, or reason for failures either. we probably should graph pass/fail as well as each of the common reasons for failures.